pkg/collector/git_collector.go

```diff
@@ -154,7 +154,7 @@ func (c *GitCollector) CollectData(data []byte, outputPath string) (string, erro
 	if err != nil {
 		return u, err
 	}
-	err = gitClient.Push(ghPagesDir, "origin", false, false, "HEAD")
+	err = gitClient.Push(ghPagesDir, "origin", false, false, "HEAD:"+c.gitBranch)
```

Comment:
Provided you are on the gh-pages branch `git push HEAD` is just a shortcut for `git push gh-pages:gh-pages`. Not sure whether this makes a difference here unless there is more to it than meets the eye.